src/Uno.UI/UI/Xaml/Controls/Picker/Picker.iOS.cs

```diff
@@ -84,7 +84,7 @@ private void OnItemSourceChanged(object collection, NotifyCollectionChangedEvent
 			Items = (collection as IEnumerable)?.ToObjectArray() ?? new object[0];
 			ReloadAllComponents();
 
-			if (!Items.Contains(SelectedItem))
+			if (Items[0] != null && !Items.Contains(SelectedItem))
```

Comment:
This is going to throw an exception if Items is empty. 

Also, I really can't tell what it's intended to achieve.

Reply:
If collection is empty first items[0] is set to null but items is never null, therefore there will never be an exception thrown.

The behaviour is that when a combobox is initialized the initial selected item is isnt the first item in the collection. Like in other platforms

Reply:
Can you show me where in the code it's ensured that `Items` is never empty? I can see it's initialized as `{ null }`, but it seems like all bets are off after `ItemsSource` changes.

I'm guessing the regression appeared because this line was recently removed:

```csharp
Items = new[] { (object)null }.Concat(items).ToArray(); // ensure there's always a null present to allow deselection
```

Did you confirm with @dr1rrb that it wasn't removed accidentally?

Reply:
As discussed yesterday, yes it was intentional because it was allowing the user to select `null` which is not possible on UWP (and maybe even erasing the selected value on loaded, not sure, to validate). If we want to avoid auto select of the first value, we should insert a `null` only until a selection is made, and then remove it.